core/src/main/kotlin/entity/interaction/OptionValue.kt

```diff
@@ -0,0 +1,102 @@
+package dev.kord.core.entity.interaction
+
+import dev.kord.common.entity.CommandArgument
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.behavior.interaction.GlobalInteractionBehavior
+import dev.kord.core.entity.*
+import dev.kord.core.entity.channel.ResolvedChannel
+
+public sealed interface OptionValue<out T> {
+    public val value: T
+    public val focused: Boolean
+}
+
+public sealed interface ResolvableOptionValue<out T> : OptionValue<Snowflake> {
+    public val resolvedObject: T?
+}
```

Comment:
Changing the `value` of a `ResolvableOptionValue` to always being a `Snowflake` will cause the current implementation of `InteractionCommand.filterOptions()` to be incorrect for these types of `OptionValue`s since it does a `value.value is T` check.
I included a fix in #538.